modules/i18n/server/controllers/backend.server.controller.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+/**
+ * We want to process only one unsaved translation at the time.
+ * Otherwise the output files will be inconsistent.
+ * We make a promise chain: The next promise will start executing after the previous one finished
+ */
+const queue = { promise: Promise.resolve() };
+function enqueue(func, ...args) {
+  queue.promise = queue.promise.then(() => func(...args));
+}
+
+/**
+ * Object property order matters since ES2015
+ * So we can sort it alphabetically by keys
+ */
+const sortObject = object => Object.fromEntries(Object.entries(object).sort(([a], [b]) => a > b ? 1 : -1));
+
+/**
+ * Here we execute the standard request, not concerned with waiting
+ */
+async function processRequest(req, res) {
+  // get the needed variables
+  const { lng, ns } = req.params;
+  const [key] = Object.keys(req.body).filter(key => key !== '_t');
+  const value = (lng === 'en') ? req.body[key] : '';
+
+  const file = path.resolve(`./public/locales/${lng}/${ns}.json`);
+
+  try {
+    // create file if it doesn't exist
+    await fs.ensureFile(file);
+
+    // read current translations or set a default
+    const rawContent = await fs.readFile(file, 'utf8') || '{}';
+    const content = JSON.parse(rawContent);
+
+    const newContent = sortObject({ [key]: value, ...content });
```

Comment:
Would lodash `sortBy` simplify this? https://lodash.com/docs/4.17.15#sortBy

Reply:
Thank you for pointing out potentially simpler solutions.

Unfortunately `sortBy` won't help us here. The function works on objects, but seems to take only values into consideration, and returns an array. And we want to sort by keys.

If I misunderstood please kindly help me understand. :slightly_smiling_face: